### PR TITLE
feat: support non-Debian distros in install

### LINKED ISSUE #604: [Bug]: Installation on non-debian-based os result in failure
### Affected component

Installation with requirements/install.sh

### Bug description

Installation via commandline `bash requirements/install.sh embodied --model openpi --env maniskill_libero --venv openpi-libero-venv` result in an error `apt-get could not be found. This script is intended for Debian-based systems.`. The full outputs are provided in logs.

My server is a AlmaLinux OS (9.5), which is indeed not a debian-based system. More information of the system is provided in "Other reproduction info" section. It has no docker installation as well.

Thank you!

### Configuration YAML file

N/A

### Log file

❯ bash requirements/install.sh embodied --model openpi --env maniskill_libero --venv openpi-libero-venv
Using CPython 3.11.14
Creating virtual environment at: openpi-libero-venv
✔ A virtual environment already exists at `openpi-libero-venv`. Do you want to replace it? · yes
Activate with: source openpi-libero-venv/bin/activate
Resolved 404 packages in 7ms
░░░░░░░░░░░░░░░░░░░░ [0/161] Installing wheels...                                                                                                     warning: Failed to hardlink files; falling back to full copy. This may lead to degraded performance.
         If the cache and target directories are on different filesystems, hardlinking may not be supported.
         If this is intentional, set `export UV_LINK_MODE=copy` or use `--link-mode=copy` to suppress this warning.
Installed 161 packages in 22.98s
 + absl-py==2.3.1
 + accelerate==1.12.0
 + aiohappyeyeballs==2.6.1
 + aiohttp==3.13.3
 + aiohttp-cors==0.8.1
 + aiosignal==1.4.0
 + annotated-types==0.7.0
 + antlr4-python3-runtime==4.11.1
 + argcomplete==3.6.3
 + attrs==25.4.0
 + boto==2.49.0
 + cachetools==5.5.2
 + certifi==2026.1.4
 + cffi==2.0.0
 + cfgv==3.5.0
 + charset-normalizer==3.4.4
 + click==8.3.1
 + colorful==0.6.0a1
 + crcmod==1.7
 + cryptography==43.0.3
 + datasets==3.6.0
 + deprecated==1.3.1
 + dill==0.3.8
 + distlib==0.4.0
 + einops==0.8.1
 + fasteners==0.20
 + filelock==3.20.3
 + frozenlist==1.8.0
 + fsspec==2025.3.0
 + gcs-oauth2-boto-plugin==3.3
 + gitdb==4.0.12
 + gitpython==3.1.46
 + google-api-core==2.29.0
 + google-apitools==0.5.35
 + google-auth==2.39.0
 + google-auth-httplib2==0.3.0
 + google-reauth==0.1.1
 + googleapis-common-protos==1.72.0
 + grpcio==1.78.0rc2
 + gsutil==5.35
 + hf-xet==1.2.1rc0
 + httplib2==0.20.4
 + huggingface-hub==0.36.0
 + hydra-core==1.4.0.dev1
 + icmplib==3.0.4
 + identify==2.6.16
 + idna==3.11
 + importlib-metadata==8.0.0
 + iniconfig==2.3.0
 + jinja2==3.1.6
 + jsonschema==4.26.0
 + jsonschema-specifications==2025.9.1
 + latex2sympy2==1.9.0 (from git+https://ghfast.top/github.com/RLinf/latex2sympy2.git@35fa1005d4cd9149d08c0b6f7efb233e4bbd25f7)
 + liger-kernel==0.6.4
 + markdown==3.10
 + markdown-it-py==4.0.0
 + markupsafe==3.0.3
 + mdurl==0.1.2
 + monotonic==1.6
 + mpmath==1.3.0
 + msgpack==1.1.2
 + multidict==6.7.0
 + multiprocess==0.70.16
 + networkx==3.6.1
 + ninja==1.13.0
 + nodeenv==1.10.0
 + numpy==2.2.6
 + nvidia-cublas-cu12==12.4.5.8
 + nvidia-cuda-cupti-cu12==12.4.127
 + nvidia-cuda-nvrtc-cu12==12.4.127
 + nvidia-cuda-runtime-cu12==12.4.127
 + nvidia-cudnn-cu12==9.1.0.70
 + nvidia-cufft-cu12==11.2.1.3
 + nvidia-curand-cu12==10.3.5.147
 + nvidia-cusolver-cu12==11.6.1.9
 + nvidia-cusparse-cu12==12.3.1.170
 + nvidia-cusparselt-cu12==0.6.2
 + nvidia-ml-py==13.580.82
 + nvidia-nccl-cu12==2.21.5
 + nvidia-nvjitlink-cu12==12.4.127
 + nvidia-nvtx-cu12==12.4.127
 + nvitop==1.6.1
 + oauth2client==4.1.3
 + omegaconf==2.4.0.dev4
 + opencensus==0.11.4
 + opencensus-context==0.2.dev0
 + opentelemetry-api==1.26.0
 + opentelemetry-exporter-prometheus==0.47b0
 + opentelemetry-proto==1.39.1
 + opentelemetry-sdk==1.26.0
 + opentelemetry-semantic-conventions==0.47b0
 + packaging==26.0rc3
 + pandas==3.0.0rc2
 + pillow==12.1.0
 + pip==25.3
 + platformdirs==4.5.1
 + pluggy==1.6.0
 + pre-commit==4.5.1
 + prettytable==3.17.0
 + prometheus-client==0.24.1
 + propcache==0.4.1
 + proto-plus==1.27.0
 + protobuf==6.33.4
 + psutil==7.2.1
 + py-spy==0.4.1
 + pyarrow==22.0.0
 + pyasn1==0.6.1
 + pyasn1-modules==0.4.2
 + pybind11==3.0.1
 + pycparser==2.23
 + pydantic==2.12.5
 + pydantic-core==2.41.5
 + pyecharts==2.0.9
 + pygments==2.19.2
 + pylatexenc==3.0a33
 + pyopenssl==24.2.1
 + pyparsing==3.3.1
 + pyrealsense2==2.56.5.9235
 + pytest==9.0.2
 + python-dateutil==2.9.0.post0
 + pyu2f==0.1.5
 + pyyaml==6.0.3
 + ray==2.47.1
 + referencing==0.37.0
 + regex==2026.1.15
 + requests==2.32.5
 + retry-decorator==2.0a1
 + rich==13.9.4
 + rpds-py==0.30.0
 + rsa==4.7.2
 + ruff==0.14.13
 + safetensors==0.7.0
 + scipy==1.17.0
 + sentencepiece==0.2.1
 + sentry-sdk==3.0.0a7
 + setuptools==75.8.2
 + simplejson==3.20.2
 + six==1.17.0
 + smart-open==7.5.0
 + smmap==5.0.2
 + swanlab==0.7.6
 + sympy==1.13.1
 + tensorboard==2.20.0
 + tensorboard-data-server==0.7.2
 + torch==2.6.0
 + torch-memory-saver==0.0.9
 + torchdata==0.11.0
 + tqdm==4.67.1
 + triton==3.2.0
 + typing-extensions==4.15.0
 + typing-inspection==0.4.2
 + urllib3==2.6.3
 + virtualenv==20.36.1
 + wandb==0.24.0
 + wcwidth==0.2.14
 + werkzeug==3.1.5
 + word2number==1.1
 + wrapt==2.1.0.dev1
 + xxhash==3.6.0
 + yarl==1.22.0
 + zipp==3.23.0
Resolved 404 packages in 1ms
Uninstalled 1 package in 5ms
░░░░░░░░░░░░░░░░░░░░ [0/106] Installing wheels...                                                                                                     warning: Failed to hardlink files; falling back to full copy. This may lead to degraded performance.
         If the cache and target directories are on different filesystems, hardlinking may not be supported.
         If this is intentional, set `export UV_LINK_MODE=copy` or use `--link-mode=copy` to suppress this warning.
Installed 106 packages in 26.37s
 + addict==2.4.0
 + anyio==4.12.1
 + array-record==0.8.3
 + astunparse==1.6.3
 + av==16.1.0
 + bddl==3.6.0
 + blinker==1.9.0
 + cloudpickle==3.1.2
 + configargparse==1.7.1
 + contourpy==1.3.3
 + cycler==0.12.1
 + dash==4.0.0rc6
 + distro==1.9.0
 + dm-tree==0.1.9
 + docstring-parser==0.17.0
 + donfig==0.8.1.post1
 + draccus==0.11.5
 + easydict==1.13
 + etils==1.13.0
 + evdev==1.9.2
 + fastjsonschema==2.21.2
 + flask==3.1.2
 + flatbuffers==25.12.19
 + fonttools==4.61.1
 + future==1.0.0
 + gast==0.7.0
 + glfw==2.10.0
 + google-crc32c==1.8.0
 + google-pasta==0.2.0
 + gym==0.26.2
 + gym-notices==0.1.0
 + h11==0.16.0
 + h5py==3.15.1
 + httpcore==1.0.9
 + httpx==0.28.1
 + imageio==2.37.2
 + imageio-ffmpeg==0.6.0
 + immutabledict==4.2.2
 + importlib-resources==6.5.2
 + itsdangerous==2.2.0
 + jiter==0.12.0
 + joblib==1.5.3
 + jupyter-core==5.9.1
 + jupytext==1.19.0.dev0
 + keras==3.13.1
 + kiwisolver==1.4.10rc0
 + libclang==18.1.1
 + llvmlite==0.44.0
 + lxml==6.0.2
 + matplotlib==3.10.8
 + mdit-py-plugins==0.5.0
 + mergedeep==1.3.4
 + ml-dtypes==0.5.4
 + mujoco==3.4.0
 + mypy-extensions==1.1.0
 + namex==0.1.0
 + narwhals==2.15.0
 + nbformat==5.10.4
 + nest-asyncio==1.6.0
 + nltk==3.9.2
 + numba==0.61.2
 + numcodecs==0.16.5
 + open3d==0.19.0
 + openai==2.15.0
 + opencv-python==4.12.0.88
 + openexr==3.4.4
 + opt-einsum==3.4.0
 + optree==0.18.0
 + orjson==3.11.5
 - packaging==26.0rc3
 + packaging==25.0
 + peft==0.11.1
 + plotly==6.5.2
 + promise==2.3
 + pynput==1.8.1
 + pyopengl==3.1.10
 + pyperclip==1.11.0
 + pyquaternion==0.9.9
 + python-xlib==0.33
 + pyvers==0.1.0
 + pyyaml-include==1.4.1
 + retrying==1.4.2
 + robosuite==1.4.1
 + sapien==3.0.1
 + scikit-learn==1.8.0
 + simple-parsing==0.1.7
 + sniffio==1.3.1
 + tensordict==0.10.0
 + tensorflow==2.20.0
 + tensorflow-addons==0.23.0
 + tensorflow-datasets==4.9.9
 + tensorflow-graphics==2021.12.3
 + tensorflow-metadata==1.17.3
 + termcolor==3.3.0
 + threadpoolctl==3.6.0
 + timm==0.9.10
 + tokenizers==0.19.1
 + toml==0.10.2
 + torchvision==0.21.0
 + traitlets==5.14.3
 + transformers==4.40.1
 + transforms3d==0.4.2
 + trimesh==4.11.0
 + typeguard==2.13.3
 + typing-inspect==0.9.0
 + wheel==0.45.1
 + zarr==3.1.5
apt-get could not be found. This script is intended for Debian-based systems.

### Environment

**python -V:**
Python 3.9.21
**uv pip list:**
(empty)
**nvidia-smi:**
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 575.51.03              Driver Version: 575.51.03      CUDA Version: 12.9     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA L40S                    On  |   00000000:A1:00.0 Off |                    0 |
| N/A   26C    P8             35W /  350W |       0MiB /  46068MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+
                                                                                         
+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI              PID   Type   Process name                        GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|  No running processes found                                                             |
+-----------------------------------------------------------------------------------------+
**RLinf version:**
main branch on Jan 20, 2026
**Docker image tag:** 
N/A (using local installation)

### Other reproduction info

\> cat /etc/os-release        
NAME="AlmaLinux"
VERSION="9.5 (Teal Serval)"
ID="almalinux"
ID_LIKE="rhel centos fedora"
VERSION_ID="9.5"
PLATFORM_ID="platform:el9"
PRETTY_NAME="AlmaLinux 9.5 (Teal Serval)"
ANSI_COLOR="0;34"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:almalinux:almalinux:9::baseos"
HOME_URL="https://almalinux.org/"
DOCUMENTATION_URL="https://wiki.almalinux.org/"
BUG_REPORT_URL="https://bugs.almalinux.org/"

ALMALINUX_MANTISBT_PROJECT="AlmaLinux-9"
ALMALINUX_MANTISBT_PROJECT_VERSION="9.5"
REDHAT_SUPPORT_PRODUCT="AlmaLinux"
REDHAT_SUPPORT_PRODUCT_VERSION="9.5"
SUPPORT_END=2032-06-01

### Before submitting a new issue...

- [x] Have you checked relevant issues, FAQs (https://rlinf.readthedocs.io/en/latest/rst_source/faq.html), or asked the chatbot at the top right corner of the [documentation page](https://rlinf.readthedocs.io).

### PR DESCRIPTION
**Pull request by Wheeeeeeeeels**
## Summary
- Add support for RHEL/CentOS/Fedora/AlmaLinux (dnf/yum) and Arch Linux (pacman) package managers
- Fix package name mappings for non-Debian systems
- Handle curl-minimal conflicts with --allowerasing flag
- Enable EPEL and CRB repositories for RHEL-based systems

## Test Plan
- [x] Tested on AlmaLinux 9 using Docker container
- [x] All 295 packages installed successfully
- [x] Rendering runtime configuration files created correctly

Fixes #604

### CODE REVIEW

**Comment by zoeyuchao** [comment]
@Wheeeeeeeeels Hi, thanks for your contribution! the pr title doesn't pass the check, could u change the Debian to debian?

**Comment by andylin-hao** [comment]
Hi @Wheeeeeeeeels, thanks a lot for the support! I just made some improvements on the root privilege detection and CI tests for these distros here: https://github.com/Wheeeeeeeeels/RLinf/pull/1. Could you check and merge it into your branch? Thanks!

**Comment by andylin-hao** [comment]
Hi @Wheeeeeeeeels, there is a CI failure (unrelated to your changes) due to a recent update of the `wheel` package, causing it to be incompatible with an older version of setuptools. I've fixed it in https://github.com/Wheeeeeeeeels/RLinf/pull/2. Could you check and merge it as well? Thanks!
